### ISSUE TITLE
mapAsyncUnordered sometimes hangs after exceptions

### ISSUE THREAD

**Issue by renecouto** [typelevel/fs2#2726]
The following example will throw a timeoutException if we use mapAsyncUnordered, but will complete when using mapAsync

Without the Async[IO].timeout, when using mapAsyncUnordered, the code would hang forever
```
import fs2.Stream
def divideByZerorEvery25Elements() = Stream.range(25, -2, -1).repeat.covary[IO].mapAsyncUnordered(20)(x => IO(1/x)).compile.drain
for (n <- 1 to 500) {
  try {
    Async[IO].timeout(divideByZerorEvery25Elements(), 2.seconds).unsafeRunSync()
  } catch {
    case e: TimeoutException =>
      println(s"on iteration ${n}")
      throw e
    case _: Throwable => ()
  }
}
```
Using:  fs2-core 3.2.2 and 3.1.5, scala 2.13.3 and 2.13.6

Since scastie times out after 30 seconds, I recommend trying it in your pc
https://scastie.scala-lang.org/scfMzh2rToKpIKud0uJe2w

**Comment by nikiforo** [comment]
Although it hangs in `main`, it doesn't in #2627.
My suspicion is that `forkOnElem` isn't guarded with uncancellable. I'll make a test and add it to the PR in a day.

**Comment by nikiforo** [comment]
@renecouto, can you confirm that the issue is resolved in v3.2.3?

**Comment by renecouto** [comment]
beautiful! Issued resolved
